corehq/messaging/tasks.py

```diff
@@ -147,22 +147,16 @@ def run_messaging_rule(domain, rule_id):
     rule = _get_cached_rule(domain, rule_id)
     if not rule:
         return
-
     progress_helper = MessagingRuleProgressHelper(rule_id)
+    total_cases_count = CaseES().domain(domain).case_type(rule.case_type).count()
+    progress_helper.set_total_cases_to_be_processed(total_cases_count)
 
     def _run_rule_sequentially():
-        incr = 0
         progress_helper.set_initial_progress()
         for case_id in get_case_ids_for_messaging_rule(domain, rule.case_type):
             sync_case_for_messaging_rule.delay(domain, case_id, rule_id)
-            incr += 1
-            if incr >= 1000:
-                progress_helper.increase_total_case_count(incr)
-                incr = 0
-                if progress_helper.is_canceled():
-                    break
-
-        progress_helper.increase_total_case_count(incr)
+            if progress_helper.is_canceled():
```

Comment:
Hey @mkangia, 
I was thinking we can check if the process is canceled in batches as it was happening earlier, the same can be implemented for `run_messaging_rule_for_shard`.
It will significantly reduce calls to redis.

Reply:
Good catch! ya this will hit redis 1000 times more now. We surely don't want that.
Earlier we were checking after 1000 cases, we can do the same to avoid changing functionality and keep it out of scope for this PR. You can check out the use of chunked in "run_messaging_rule_for_shard"
Plus we would need to update that method too, infact that is the one that is used almost always.

Reply:
Yep, will do that in a separate PR.

Reply:
Sorry, i edited my comment poorly. This should be handled in this PR itself and not as a follow up.